demo/plugins/lookup_plugins/conjur_variable.py

```
@@ -0,0 +1,156 @@
+from base64 import b64encode
+from httplib import HTTPConnection
+from netrc import netrc
+from os import environ
+from sys import stderr
+from time import time
+from urllib import quote_plus
+from urlparse import urlparse
+
+
+def exit_error(err):
+    stderr.write(err)
+    exit(1)
+
+
+class Token:
+    def __init__(self, http_connection, id, api_key, account):
+        self.http_connection = http_connection
+        self.id = id
+        self.api_key = api_key
+        self.token = None
+        self.refresh_time = 0
+        self.account = account
+
+    # refresh
+    # Exchanges API key for an auth token, storing it base64 encoded within the
+    # 'token' member variable. If it fails to obtain a token, the process exits.
+    def refresh(self):
+        authn_url = '/authn/%s/%s/authenticate' % (quote_plus(self.account), quote_plus(self.id))
+        self.http_connection.request('POST', authn_url, self.api_key)
+
+        response = self.http_connection.getresponse()
+
+        if response.status != 200:
+            exit_error('Failed to authenticate as \'%s\'' % (self.id))
+
+        self.token = b64encode(response.read())
+        self.refresh_time = time() + 5 * 60
+
+    # get_header_value
+    # Returns the value for the Authorization header. Refreshes the auth token
+    # before returning if necessary.
+    def get_header_value(self):
+        if time() >= self.refresh_time:
+            self.refresh()
+
+        return 'Token token="%s"' % (self.token)
+
+
+def load_conf(conf_path):
+    conf_path = os.path.expanduser(conf_path)
+
+    if not os.path.isfile(conf_path):
+        return {}
+
+    with open(conf_path, 'r') as conf_file:
+        try:
+            return yaml.load(conf_file)
+        except yaml.YAMLError as e:
+            exit_error(e)
+
+
+def load_identity(identity_path, appliance_url):
+    identity_path = os.path.expanduser(identity_path)
+
+    if not os.path.isfile(identity_path):
+        return {}
```

What happens when someone tries to use this module without an identity?

This code makes it seem like it will silently fail, when instead we want it to blow up and stop the Ansible run.

---

Not quite. Look at line 117 (which calls this function) and you'll see that we let the user to have the identity in one of 3 places: /etc/, ~/ and env variables (btw, this was already in the v4 plugin, i didn't change this). 

So, if the function above doesn't get an identity file it's ok as the identity may lie in one of the other places. Only if all 3 don't have an identity the script will exit with an error (in line 126)

---

I'll add a comment about this